src/UI/CO_InfoController.java

```diff
@@ -0,0 +1,109 @@
+package UI;
+
+import UI.InputHandler.InputAction;
+import CommandingOfficers.Commander;
+import CommandingOfficers.CommanderInfo;
+
+import java.util.ArrayList;
+
+import CommandingOfficers.COMaker.InfoPage;
+import Engine.GameInstance;
+import Engine.OptionSelector;
+
+public class CO_InfoController implements InfoController
+{
+  private ArrayList<CommanderInfo> coInfos;
+  private GameInstance myGame;
+  
+  private OptionSelector coOptionSelector;
+  private OptionSelector[] pageSelectors;
+
+  public CO_InfoController( GameInstance game )
+  {
+    myGame = game;
+    ArrayList<CommanderInfo> infos = new ArrayList<CommanderInfo>();
+    
+    for( Commander co : myGame.commanders )
+    {
+      infos.add(co.coInfo);
+    }
+    
+    init(infos);
+  }
+
+  public CO_InfoController( ArrayList<CommanderInfo> infos )
+  {
+    init(infos);
+  }
+  
+  private void init( ArrayList<CommanderInfo> infos )
+  {
+    coInfos = infos;
+    
+    coOptionSelector = new OptionSelector(coInfos.size());
+    pageSelectors = new OptionSelector[coInfos.size()];
+    for( int i = 0; i < coInfos.size(); ++i )
+    {
+      pageSelectors[i] = new OptionSelector(coInfos.get(i).maker.infoPages.size());
+    }
+  }
+  
+  @Override
+  public boolean handleInput( InputAction action )
+  {
+    boolean goBack = false;
+    switch( action )
+    {
+      case DOWN:
+      case UP:
+        // Up/Down changes which CO has focus.
+        coOptionSelector.handleInput( action );
+        break;
+      case LEFT:
+      case RIGHT:
```

Comment:
I feel like it might make sense to keep the different pages in sync. Like, If I'm on Strong and cycle over to his Passive page, and then move down to Patch, it should also show his Passive page instead of jerking me back over to the info page.

We'd still have to figure out how to handle COs with more than two abilities though. Maybe have a little navigation pane that shows a "map" of the info screens so you are always oriented.

Reply:
Hmm, that's fair. Making the UI for that sounds hard, though.
Also, I have a mild ambition to make it one, contiguous image that you can scroll down, at some point, but that hasn't happened and probably won't soon.